gcp-gmp-integration.md

```diff
@@ -68,4 +68,65 @@ From your [GCP Monitoring - Metrics explorer console](https://console.cloud.goog
 avg(node_cpu_hourly_cost) by (cluster_id)
 ```
 
-Read our [Custom Prometheus integration troubleshooting guide](custom-prom.md#troubleshooting) if you run into any errors while setting up the integration. For support from GCP, you can submit a support request at the [GCP support hub](https://cloud.google.com/support-hub).
+## Troubleshooting
+
+### Cluster efficiency displaying as 0%, or efficieny only displaying for most recent cluster
+
+The below queries must return data for Kubecost to calculate costs correctly. For the queries to work, set the environment variables:
+
+```bash
+KUBECOST_NAMESPACE=kubecost
+KUBECOST_DEPLOYMENT=kubecost-cost-analyzer
+CLUSTER_ID=YOUR_CLUSTER_NAME
+```
+
+1. Verify connection to GMP and that the metric for `container_memory_working_set_bytes` is available:
+
+If you have set `kubecostModel.promClusterIDLabel`, you will need to change the query (`CLUSTER_ID`) to match the label.
+
+```bash
+kubectl exec -i -t -n $KUBECOST_NAMESPACE \
+  deployments/$KUBECOST_DEPLOYMENT -c cost-analyzer-frontend \
+  -- curl "0:9090/model/prometheusQuery?query=container_memory_working_set_bytes\{CLUSTER_ID=\"$CLUSTER_ID\"\}" \
+ |jq
+```
+
+2. Verify Kubecost metrics are available in GMP:
+
+```bash
+kubectl exec -i -t -n $KUBECOST_NAMESPACE \
+  deployments/$KUBECOST_DEPLOYMENT -c cost-analyzer-frontend \
+  -- curl "0:9090/model/prometheusQuery?query=node_total_hourly_cost\{CLUSTER_ID=\"$CLUSTER_ID\"\}" \
+ |jq
+```
+
+If the above queries fail, check the following:
+
+1. Check logs of the `sigv4proxy` container (may be the Kubecost deployment or Prometheus Server deployment depending on your setup):
```

Comment:
@kc-adawson Do you have this running like described above?
GMP doesn't call it a sigV4 proxy? Looks like it is `gke.gcr.io/prometheus-engine/frontend:v0.4.1-gke.0`
Is there any other advice you have here for troubleshooting?

Reply:
Just catching up on this - In my sandbox, I'm setting GMP up in the way documented here. https://docs.kubecost.com/install-and-configure/install/custom-prom/gcp-gmp-integration .

In my setup, the "id" metric that would be retrieved via these debug instructions returns blank, so if people run into an "id" that is blank using the debug steps,  they can use the following to force set "cluster" as the Prom Cluster ID label

kubecostModel:
  promClusterIDLabel: cluster
  
  ```
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "__name__": "",
          "cluster": "",
          "id": "/",         <<<<--- This is blank for GMP/(Autopilot?) Prometheus, need to use "cluster"
          "instance": "",
          "job": "kubelet",
          "location": "",
          "node": "",
          "project_id": ""
        },
        "value": [
          1697627020,
          "2358820864"
        ]
      },
```

Reply:
Thanks, I've added it into the current doc.